CONTRIBUTING.md

```diff
@@ -1,53 +1,307 @@
+
+When issues are closed, a comment is added about why.
+Closing issues is a reversible action, and it is always acceptable to comment and explain (politely) why an issue should not have been closed.
+
+### Labels
+
+We make extensive use of labels.
+The main categories are:
 
-### Using forks
+- [Housekeeping](https://github.com/OAI/OpenAPI-Specification/labels/Housekeeping) for meetings, project logistics, etc.
+- [approved pr port](https://github.com/OAI/OpenAPI-Specification/labels/approved%20pr%20port) for pull requests that repeat a change from one version to another
+- most other tags are used to group similar or related issues into topic areas; this list is ever-changing
 
-All work **MUST be done on a fork**, using a branch from the _earliest relevant and [active](#active-branches)_ `vX.Y-dev` branch, and then submitted as a PR to that `vX.Y-dev` branch.
-For example, if a change in November 2024 apples to both 3.1 and 3.2, the PR would go to the `v3.1-dev` branch, which will be merged up to `v3.2-dev` before the next 3.2 release.
+Labels related to [issue automation](#appendix-issue-automation)
+
+- [Needs attention](https://github.com/OAI/OpenAPI-Specification/labels/Needs%20attention) automated tag when an issue is updated
+- [Needs author feedback](https://github.com/OAI/OpenAPI-Specification/labels/Needs%20author%20feedback) used to indicate that more information is needed from the issue creator
+- [No recent activity](https://github.com/OAI/OpenAPI-Specification/labels/No%20recent%20activity) if no information is received, the issue is marked for closure (automatic after 30 days)
+
+### Milestones
+
+We use milestones in GitHub to plan what should be included in future releases.
+Issues and pull requests should both be added to the milestone we expect they will be released in.
+Any changes that aren't ready in time for release should be moved to the next milestone or untagged.
+
+The milestones and items assigned to them are under constant review and subject to change.
+
+### Projects
+
+The OpenAPI Initiative uses GitHub Projects to manage work _outside_ of the specification development process.  There are currently two active projects:
+
+* [Contributor Guidance](https://github.com/orgs/OAI/projects/5/views/1)
+* [Automation & Infrastructure](https://github.com/orgs/OAI/projects/4/views/1)
+
+## Pull requests
+
+> [!NOTE]
+> Since the 3.0.4 and 3.1.1 releases (October 2024), the OAS is developed in the `src/oas.md` file.
+> Check the [Branches](#branches) section for more information about the updated branching strategy.
+
+Changes to the next version of the specification are welcome and can be proposed by anyone.
+
+For large changes that will need discussion, please use the [Proposal process](#propose-a-specification-change).
+For other changes, we recommend [opening an issue](#issues) first, so that you can get some feedback and any extra input you need before spending a lot of time on something.
```

Comment:
Should GitHub discussions be mentioned there?  Earlier parts of this document imply that big things go to discussions, actionable thing go to issues, and the formal proposal process seems to be a way to bridge the gap by formalizing a discussion that might not be ready for individual issues yet.  At least that's what I get from it?

Reply:
I've structured the mentions of discussions in the document something like this:
 - under "how to contribute": use discussions to give us some feedback or answer a question
 - in the discussions section, all about discussions
 - in the issues section: for big changes, use a proposal
 - in this section (pull requests): for big changes, use a proposal
 - in the proposals section: start the proposal process with a discussion

I'm quite comfortable with the structure as it is, trying to base it around a pyramid of user engagement where first they are visiting, then they are participating, then they are contributing, then they are re-engineering. I'm not wedded to it though if there's somewhere else we should mention it! I'm also not too worried if people open an issue when they should have used a discussion - engagement is good and there's nothing irreversible!